src/pip/_internal/pep425tags.py

```diff
@@ -252,10 +252,18 @@ def get_supported(versions=None, noarch=False, platform=None,
         abis[0:0] = [abi]
 
     abi3s = set()
-    import imp
-    for suffix in imp.get_suffixes():
-        if suffix[0].startswith('.abi'):
-            abi3s.add(suffix[0].split('.', 2)[1])
+    try:
+        from importlib.machinery import EXTENSION_SUFFIXES
+
+        for suffix in EXTENSION_SUFFIXES:
+            if suffix.startswith('.abi'):
+                abi3s.add(suffix.split('.', 2)[1])
+    except ImportError:
+        import imp
+
+        for suffix in imp.get_suffixes():
```

Comment:
It would probably be better to not duplicate the loop. I suggest adding a `get_extension_suffixes()` to `pip._internal.utils.compat` that basically exposes the iterable that would be looped upon.

Reply:
Sure. Something like that?

Reply:
The try/except with an `ImportError` may be better, but I followed how other functons are defined in the file.